drivers/serial/uart_nrf5.c

```diff
@@ -305,13 +250,13 @@ static unsigned char uart_nrf5_poll_out(struct device *dev,
 	 */
 
 	/* reset transmitter ready state */
-	uart->EVENTS_TXDRDY = 0;
+	nrf_uart_event_clear(uart, NRF_UART_EVENT_TXDRDY);
```

Comment:
This has been recently changed upstream to be done **after** the `while` loop due to a race condition. Can you take a look?

Reply:
I am talking about [this change](https://github.com/zephyrproject-rtos/zephyr/commit/05c45e359a94e96353dfac1f0abeccf6893e988a#diff-fb209e20d9cf9e70b3dbd2748ae43f90) see however issue #7755, which seems to have been introduced by it. Hence I believe we need to incorporate that fix but also resolve the issue it introduces.

Reply:
It seems that the change you mentioned is included in this PR: the register used to be cleared at the end of the function, now it is cleared before. That seems to have broken the console in master (see issue [#7755](https://github.com/zephyrproject-rtos/zephyr/issues/7755)) but it seems to be working fine in this PR.

Reply:
Ah true @lemrey I misread the code.